Atomic/backends/_docker.py

```diff
@@ -191,26 +210,31 @@ def start_container(self, name):
     def stop_container(self, con_obj):
         return self.d.stop(con_obj.id)
 
-    def pull_image(self, image, pull_args):
-        # Add this when atomic registry is incorporated.
-        # if self.args.reg_type == "atomic":
-        #     pull_uri = 'atomic:'
-        # else:
-        #     pull_uri = 'docker://'
-        img_obj = self._make_remote_image(image)
-        fq_name = img_obj.fq_name
-        insecure = True if util.is_insecure_registry(self.d.info()['RegistryConfig'], util.strip_port(img_obj.registry)) else False
-
-        # This needs to be re-enabled with Aaron's help
+    def pull_image(self, image, **kwargs):
+        debug = kwargs.get('debug', False)
+        if image.startswith("dockertar:"):
+            path = image.replace("dockertar:", "", 1)
+            with open(path, 'rb') as f:
+                self.d.load_image(data=f)
+            return 0
+        remote_image = self.make_remote_image(image)
+        fq_name = remote_image.fq_name
+        local_image = self.has_image(image)
+        if local_image is not None:
+            if self.already_has_image(local_image, remote_image):
+                raise ValueError("Latest version of {} already present.".format(image))
+        registry, _, _, tag, _ = util.Decompose(fq_name).all
+        image = "docker-daemon:{}".format(image)
+        if not image.endswith(tag):
+            image += ":{}".format(tag)
+        insecure = True if util.is_insecure_registry(self.d.info()['RegistryConfig'], util.strip_port(registry)) else False
         trust = Trust()
-        trust.set_args(pull_args)
+        #trust.set_args(pull_args)
```

Comment:
Aaron and I speculated that these are no longer needed.  I commented it out to test it.  Just need to delete them.  Will do now.